server/server.go

```diff
@@ -96,12 +99,14 @@ func NewServer(address string, handler Handler, opts Options) Server {
 	scope := instrumentOpts.MetricsScope()
 
 	s := &server{
-		address:    address,
-		opts:       opts,
-		log:        instrumentOpts.Logger(),
-		closedChan: make(chan struct{}),
-		metrics:    newServerMetrics(scope),
-		handler:    handler,
+		address:                address,
+		opts:                   opts,
```

Comment:
Still used for report interval, but I can store it separately as well and get rid of opts.